### PR TITLE
Feature/add ignore sass files

### PR DESCRIPTION
**Pull request by grafikchaos**
### Problem:

We use Foundation 6 installed via a composer package (`zurb/foundation`) for our base theme (and `Snowdog/blank` as it's parent). When we add the Zurb_Foundation to the theme's `modules` section we are experiencing compilation errors because the inheritance engine is a little too aggressive in gathering anything from anywhere that ends with `.scss` (i.e., Foundation's `docs` directory doesn't need to be compiled, just ignored). 

We've tried bypassing the `docs` directory specifying the path to the module as such:

```json
"modules": {
    "Zurb_Foundation": "vendor/zurb/foundation/scss"
},
```

but that leads to compilation errors due to how foundation imports files from outside of the `scss` directory.

### Proposed Solution:

The idea is that if a theme developer wants to exclude certain paths from being compiled (i.e., Zurb Foundation's 'docs' directory) that they can add the `"ignore": [ "!**/docs/**/*.scss" ]` section to the `themes.json` configuration and when all the symlinks are generated the directories/files that match those patterns will not be created and thus excluding them from being compiled.

Example `themes.json` from our multi-store project with a theme hierarchy:

```json
{
  "snowdog-blank": {
    "src": "vendor/snowdog/theme-blank-sass",
    "dest": "pub/static/frontend/Snowdog/blank",
    "locale": ["en_US"]
  },
  "ash-wampa": {
    "src": "vendor/augustash/theme-frontend-wampa",
    "dest": "pub/static/frontend/Ash/wampa",
    "locale": ["en_US"],
    "postcss": ["plugins.autoprefixer()"],
    "parent": "snowdog-blank",
    "modules": {
      "Zurb_Foundation": "vendor/zurb/foundation"
    },
    "ignore": [ "!**/docs/**/*.scss"]
  },
  "hdis-default": {
    "src": "app/design/frontend/HDIS/default",
    "dest": "pub/static/frontend/HDIS/default",
    "locale": ["en_US"],
    "parent": "ash-wampa",
    "postcss": ["plugins.autoprefixer()"],
    "modules": {
      "Zurb_Foundation": "vendor/zurb/foundation",
      "Ash_Wampa": "vendor/augustash/theme-frontend-wampa"
    },
    "ignore": [ "!**/docs/**/*.scss" ]
  },
  "hdis-interdry": {
      "src": "app/design/frontend/HDIS/default",
      "dest": "pub/static/frontend/HDIS/default",
      "locale": ["en_US"],
      "parent": "ash-wampa",
      "postcss": ["plugins.autoprefixer()"],
      "modules": {
        "Zurb_Foundation": "vendor/zurb/foundation",
        "Ash_Wampa": "vendor/augustash/theme-frontend-wampa"
      },
      "ignore": [ "!**/docs/**/*.scss" ]
  }
}
```

### CODE REVIEW

File: helper/inheritance-resolver.js [hunk 1]
```diff
@@ -11,8 +11,17 @@ module.exports = function(plugins, config, name) { // eslint-disable-line func-n
   }
 
   function generateSymlinks(src, dest, replacePattern, ignore = []) {
+    var themeIgnoreFiles = [];
+    // Iterate through the config.themes to grab any ignored paths that
+    // themers want to be excluded from the gulp.src() and sass compilation
+    Object.keys(config.themes).forEach(name => {
```
Comment (Igloczek) [inline helper/inheritance-resolver.js]: I don't like this part. Theme specific configuration shouldn't be used for all themes.

Comment (grafikchaos) [inline helper/inheritance-resolver.js]: Understandable but that was the one choke point that I could identify. Open to suggestions on how to allow directories/files to be excluded as I think that has to be considered. I originally thought the helper/scss would be good chokepoint but wasn't having any luck on getting the files to be ignored from the `gulp.src()`

Comment (Igloczek) [inline helper/inheritance-resolver.js]: But I don't get the point of taking ignores from all themes and using them for a single theme.
Even in terms of inherited themes, if a parent ignores these files, there is no need to ignore them on the child level, b/c they not exist.

IMO this part should be removed and `config.themes[name].ignore` should be used in line 24 instead `themeIgnoreFiles`.

**Comment by Igloczek** [comment]
Thanks for throwing some light on this topic and proposing the solution. I take best parts, added few own things and added to repository.
Feature will be released with next minor version.

**Pull request closed without merging**